wallet_presentation/src/commonMain/kotlin/net/thechance/mena/wallet/presentation/screen/transaction_history/TransactionFilterState.kt

```diff
@@ -16,14 +21,40 @@ data class TransactionFilterState(
     val isDateBottomSheetVisible: Boolean = false,
     val datePickerMode: DatePickerMode = DatePickerMode.START_DATE,
     val isApplyButtonLoading: Boolean = false,
-    val errorState: ErrorState? = null
+    val errorState: ErrorState? = null,
+    val oldFilterHash: Int = EMPTY_FILTER_HASH
 ) {
-    val hasActiveFilters: Boolean
-        get() = selectedTypes.isNotEmpty() || selectedStatus != FilterStatus.ALL
-                || startDate != null || endDate != null
+    val isApplyButtonEnabled: Boolean
+        get() = oldFilterHash != currentFilterHash
 
-    enum class DatePickerMode {
-        START_DATE,
-        END_DATE
+    val currentFilterHash: Int
+        get() = listOf(
+            selectedTypes.hashCode(),
+            selectedStatus.hashCode(),
+            startDate.hashCode(),
+            endDate.hashCode()
+        ).hashCode()
```

Comment:
```suggestion
    val currentFilterHash: Int
        get() = listOf(
            selectedTypes,
            selectedStatus,
            startDate,
            endDate
        ).hashCode()
```
why you don't hash the list only like this?